src/main/java/com/iwaconsolti/league/manager/service/impl/SoccerLeagueImpl.java

```diff
@@ -0,0 +1,180 @@
+package com.iwaconsolti.league.manager.service.impl;
+
+import com.iwaconsolti.league.manager.model.Matches;
+import com.iwaconsolti.league.manager.model.Players;
+import com.iwaconsolti.league.manager.model.Teams;
+import com.iwaconsolti.league.manager.service.ILeagues;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service("soccerLeague")
+@Slf4j
+public class SoccerLeagueImpl implements ILeagues {
+
+    @Value("${league.teams.limit}")
+    private int teamLimit;
+
+    private List<Players> players = new ArrayList<>();
+    private List<Teams> teams = new ArrayList<>();
+    private List<Matches> matches = new ArrayList<>();
+    private int nextPlayer = 1;
+    private int nextTeam = 1;
+    private int nextMatches = 1;
+
+    @Override
+    public Players savePlayers(Players player) {
+        log.debug("Saving player {}", player);
+        player.setId(nextPlayer++);
+        players.add(player);
```

Comment:
Before adding the player you should cehck if the team exists otherwise the player is created without a team